tests/migration/migration.go

```diff
@@ -2725,7 +2725,19 @@ var _ = SIGMigrationDescribe("VM Live Migration", decorators.RequiresTwoSchedula
 
 			By("Creating ResourceQuota with enough memory for the vmi but not enough for migration")
 			resourceQuota := newResourceQuota(resourcesToLimit, testsuite.GetTestNamespace(vmi))
-			_ = createResourceQuota(resourceQuota)
+			resourceQuota = createResourceQuota(resourceQuota)
+			Eventually(func(g gomegatypes.Gomega) error {
+				quota, err := virtClient.CoreV1().ResourceQuotas(resourceQuota.Namespace).Get(context.TODO(), resourceQuota.Name, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				for key := range resourcesToLimit {
+					g.Expect(quota.Status.Hard).To(HaveKey(key))
```

Comment:
Shouldn't you return error if the value doesn't exist

Reply:
Passing the `Gomega` and using it means that `Eventually` will continue and not fail

Reply:
>In that case why would we want to retry if g.Expect(err).To(Not(HaveOccurred())) fail?

Because we want to wait for k8s to set up the resource quotas, no?

Reply:
@xpivarc Can't we use:
```golang
for key, value := range resourcesToLimit {
   g.Expect(quota.Status.Hard).To(HaveKeyWithValue(key, value))
   g.Expect(quota.Status.Used).To(HaveKey(key))
}
```

Reply:
yes but the  `g.Expect(err).To(Not(HaveOccurred()))` above is about the get API for the quota that should always succeed iirc

Reply:
`not found`?
I mean, if that req cannot fail what is the reasoning behind this patch? I'm just trying to understand

Reply:
The quota should be found if we created it. Why would we need to retry?

Reply:
Ok, so what's the flakiness here?

Reply:
It's not about being flaky. it's about hiding errors with retries. It's not a big issue, but I don't think we should retry on network problems to me it seems like a bad habit.

Reply:
In order to stop the retry you should use `Expect` (without the `g`), but you said that you don't want a mix of usages.
What is your proposal? Maybe there are some usages I am missing :slightly_smiling_face:

Reply:
I think we should stick to one way of using `eventually` either with the `gomega.Gomega` input or without it. But of course, I'm not the one who decides, this is just my opinion

Reply:
My 2cents:
* In the vast majority of the tests we fail when there's an API error that should succeed, like this `Get()` call. I agree it's not very consistent to retry in this case.
* Passing `gomegatypes.Gomega` to the Eventually block is indeed very confusing. I think that returning an error is a much more elegant way of retrying:
```go
Eventually(func() error {
	if !strings.Contains(quota.Status.Hard, key) {
		return fmt.Errorf("quota %s is missing key %s in quota.Status.Hard: %s, quota.Name, key, quota.Status.Hard)
	}
			
	return nil
}).Should(Succeed())
```

This is much clearer, you get to format the error properly, and it's very clear what's being re-tried and what's not.

This is only a nit from my side though.

Reply:
I don't mind preferring the less gomega way